mlflow/tensorflow/__init__.py

```diff
@@ -887,14 +897,113 @@ def _patch_implementation(
+                            if isinstance(input_data[:][0], np.ndarray):
+                                is_x_supported = True
+
+                        if not is_x_supported:
+                            warnings.warn(
+                                "Tensorflow keras autologging only "
+                                "supports one of these types: numpy array, "
+                                "dictionary of (name -> numpy.ndarray), "
+                                "tf.data dataset or tf.keras.utils.Sequence"
+                            )
+                        return is_x_supported
+
+                    def _get_input_data_slice(input_example, history):
+                        input_example_slice = None
+                        if isinstance(input_example, np.ndarray):
+                            input_example_slice = input_example[:INPUT_EXAMPLE_SAMPLE_ROWS]
+                        elif isinstance(input_example, tensorflow.data.Dataset):
+                            steps = 1
+                            if history.params is not None and "steps" in history.params:
+                                steps = history.params["steps"]
+
+                            def _extract_n_steps(input_example_n_steps):
+                                if steps > 1:
+                                    return np.array(
+                                        [
+                                            v[0][:INPUT_EXAMPLE_SAMPLE_ROWS]
+                                            for v in input_example_n_steps.take(
+                                                1
+                                            ).as_numpy_iterator()
+                                        ]
+                                    )[0]
+                                return np.array(
+                                    [
+                                        v[0]
+                                        for v in input_example_n_steps.take(
+                                            INPUT_EXAMPLE_SAMPLE_ROWS
+                                        ).as_numpy_iterator()
+                                    ]
+                                )
+
+                            return _extract_n_steps(input_example)
+                        elif isinstance(input_example, dict):
+                            input_example_slice = {
+                                k: np.take(v, range(0, INPUT_EXAMPLE_SAMPLE_ROWS))
+                                for k, v in input_example.items()
+                            }
+                        elif isinstance(input_example, tensorflow.keras.utils.Sequence):
+                            input_example_slice = input_example[:][0][:INPUT_EXAMPLE_SAMPLE_ROWS]
+
+                        return input_example_slice
+
+                    def _infer_model_signature(input_example):
+                        try:
+                            model_output = history.model.predict(input_example)
```

Comment:
re: above comment, here you're referring to the encapsulated variable from the parent reference directly. Should be good to do above as well.